File: VsVim/CompletionWindowBroker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Language.Intellisense;
using Vim;

namespace VsVim
{
    [Export(typeof(ICompletionWindowBroker))]
    internal class CompletionWindowBroker : ICompletionWindowBroker
    {
        private readonly ICompletionBroker _completionBroker;
        private readonly ISignatureHelpBroker _signatureBroker;

        [ImportingConstructor]
        internal CompletionWindowBroker(
            ICompletionBroker completionBroker,
            ISignatureHelpBroker signatureBroker)
        {
            _completionBroker = completionBroker;
            _signatureBroker = signatureBroker;
        }

        bool ICompletionWindowBroker.IsCompletionWindowActive(Microsoft.VisualStudio.Text.Editor.ITextView view)
        {
            return _completionBroker.IsCompletionActive(view) || _signatureBroker.IsSignatureHelpActive(view);
        }

        void ICompletionWindowBroker.DismissCompletionWindow(Microsoft.VisualStudio.Text.Editor.ITextView view)
        {
            if (_completionBroker.IsCompletionActive(view))
            {
                _completionBroker.DismissAllSessions(view);
            }

            if (_signatureBroker.IsSignatureHelpActive(view))
            {
                _signatureBroker.DismissAllSessions(view);
            }
        }
    }
}

File: VimCore/CompletionWindowBroker.fs
#light

namespace Vim
open Microsoft.VisualStudio.Text.Editor
open Microsoft.VisualStudio.Language.Intellisense

type internal CompletionWindowBroker 
    ( 
        _textView : ITextView,
        _completionBroker : ICompletionBroker,
        _signatureBroker : ISignatureHelpBroker ) = 
    interface ICompletionWindowBroker with
        member x.TextView = _textView
        member x.IsCompletionWindowActive = 
            _completionBroker.IsCompletionActive(_textView) || _signatureBroker.IsSignatureHelpActive(_textView)
        member x.DismissCompletionWindow() = 
            if _completionBroker.IsCompletionActive(_textView) then
                _completionBroker.DismissAllSessions(_textView)
            if _signatureBroker.IsSignatureHelpActive(_textView) then
                _signatureBroker.DismissAllSessions(_textView)
